doc/requirements/io_class

```diff
@@ -19,65 +19,348 @@ It shall be possible to set a maximum size of an IO class.
+--------------------------------------------------------------------------------
+title: Export an IO class configuration to csv
+id: export_csv
+---
+
+The product shall provide an option to export current IO class configuration in
+a format compatilbe with a config file.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Load configuration from file
+id: config_file
+---
+
+The product shall provide an option to load IO class configuration from a file.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Setting an IO class configuration on loading a cache
+id: load_config
+---
+
+When a cache is loaded, IO class configuration shall be restored to the values
+used before the cache had been shut down.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Max number of IO classes
+id: max_configured
+---
+
+The product shall allow to configure up to 33 IO classes.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Attributes
+id: attributes
+---
+
+The IO class configuration shall include the following attributes for each IO class:
+
+1. Id
+2. Classification rule
+3. Eviction priority
+4. Occupancy limit
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Allowed ids
+id: id_allowed_values
+---
+
+The product shall accept the IO class ids in range 0-32 inclusive.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Allowed eviction priorities
+id: eviction_priorities_allowed_values
+---
+
+The product shall accept the eviction priorities in range 0-255 inclusive.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Empty eviction priority
+id: empty_priority
+---
+
+The product shall allow to omit eviction priority in IO class configuration.
+The IO class with empty eviction priority shall be thought of as pinned.
+
+--------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+title: Pinning IO class
+id: pinning
+---
+
+The product shall allow to pin an IO class. Data from the pinned IO class shall
+not be evicted unless its occupancy limit is not exceeded.
```

Comment:
I think stating that something should not happen in an unspecified set of circumstances makes it hard to test. Instead I would specify a set of requirements exactly describing expected eviction/replacement behavior. This would also explain the interdependencies between specific settings (pinning, eviction priority, max occupancy) which might be hard to deduce from the existing form. Let's discuss this.

Reply:
How exactly would you like to extend this req?